advance_check_void/models/account_payment.py

```diff
@@ -0,0 +1,391 @@
+            ):
+                if (
+                    payment.payment_method_code == "check_printing"
+                    and payment.is_move_sent
+                ):
+                    payment.can_be_voided = True
+                if payment.payment_method_code == "ACH-Out":
+                    payment.can_be_voided = True
+                if payment.payment_method_code == "manual" and payment.is_move_sent:
+                    payment.can_be_voided = True
+
+    @api.onchange("payment_method_id")
+    def onchange_payment_method(self):
+        """Allow to customer payment to enter check number.
+        - Based on journal check printing manual True/False for vendor."""
+        self.is_visible_check = False
+        self.is_readonly_check = False
+        if self.payment_method_id:
+            if (
+                self.payment_method_id.payment_type == "inbound"
+                and self.payment_method_code in ("check_printing", "ACH-Out", "manual")
+            ):
+                self.is_visible_check = True
+                self.check_number = 0
+            elif self._context.get("is_vendor"):
+                if (
+                    self.payment_method_id.payment_type == "outbound"
+                    and self.payment_method_code
+                    in ("check_printing", "ACH-Out", "manual")
+                ):
+                    self.is_visible_check = False
+            elif (
+                self.payment_method_id.payment_type == "outbound"
+                and self.payment_method_code in ("check_printing", "ACH-Out", "manual")
+            ):
+                self.is_visible_check = True
+                if not self.check_manual_sequencing:
+                    self.is_readonly_check = True
+
+    def print_checks(self):
+        payment_check_void_obj = self.env["payment.check.void"]
+        for rec in self:
+            if rec.payment_type == "outbound":
+                domain = [
+                    ("payment_id", "=", rec.id),
+                    ("check_number", "=", int(rec.check_number)),
+                ]
+                payment_check_void_ids = payment_check_void_obj.search(
+                    domain, order="id DESC", limit=1
+                )
+                is_manual = rec.journal_id.check_manual_sequencing
+                if payment_check_void_ids and is_manual:
+                    rec.check_number = rec.journal_id.check_next_number
+                    rec.is_visible_check = True
+                    sequence = rec.journal_id.check_sequence_id
+                    sequence.next_by_id()
+                    message = (
+                        """<ul class="o_mail_thread_message_tracking">
+                        <li>(_(Check Updated Date:)) %s</li>
+                        <li>(_(Check Number:)) %s (Generated)</li>
+                        <li>(_(State:)) %s</li></ul>"""
+                    ) % (
+                        datetime.now().strftime(DEFAULT_SERVER_DATE_FORMAT),
```

Comment:
This produces server timezone date, you probably want [user timezone date](https://www.odoo.com/documentation/14.0/developer/reference/addons/orm.html#odoo.fields.Datetime.context_timestamp).
```suggestion
                        fields.Date.to_string(fields.Date.context_timezone(self)),
```